provider/equinix/environ.go

```diff
@@ -808,11 +808,7 @@ func waitDeviceActive(ctx context.ProviderCallContext, c *packngo.Client, id str
 		Clock:    clock.WallClock,
 	})
 
-	if d != nil {
-		return d, nil
-	}
-
-	return nil, err
+	return d, err
```

Comment:
Please use `errors.Trace`, `errors.Annotate` or `errors.Annotatef`. These will ensure errors include stack traces, and are fine to call on `nil` errors, which just return nil.